### DISCUSSION TITLE
Error 401

### DISCUSSION THREAD

**Discussion by imperidal** [iTwin/itwinjs-core#2261] [Q&A] [answered]
I'm going through the iTwin Accreditation course, and trying to set up the basic viewer as per the course videos.  As soon as i hit npm run start, and logged in:

```
Error: {
                                "statusCode": 401,
                                "message": "Access denied due to invalid access_token. Make sure to provide a valid token for this API endpoint."
                            }
▼ 9 stack frames were expanded.
Function.parse
C:/Users/imper/test-itwin/node_modules/@bentley/itwin-client/src/Request.ts:167
parse
C:/Users/imper/test-itwin/node_modules/@bentley/itwin-client/src/WsgClient.ts:35
parse
C:/Users/imper/test-itwin/node_modules/@bentley/imodelhub-client/src/imodelhub/Errors.ts:90
request
C:/Users/imper/test-itwin/node_modules/@bentley/itwin-client/src/Request.ts:439
async IModelBaseHandler.getInstancesChunk
C:/Users/imper/test-itwin/node_modules/@bentley/itwin-client/src/WsgClient.ts:451
async IModelBaseHandler.getInstances
C:/Users/imper/test-itwin/node_modules/@bentley/itwin-client/src/WsgClient.ts:407
async ChangeSetHandler.get
C:/Users/imper/test-itwin/node_modules/@bentley/imodelhub-client/src/imodelhub/ChangeSets.ts:428
async Function.getLatestChangesetId
C:/Users/imper/test-itwin/node_modules/@bentley/imodeljs-frontend/src/FrontendHubAccess.ts:56
async Function.open
C:/Users/imper/test-itwin/node_modules/@bentley/imodeljs-frontend/src/CheckpointConnection.ts:60
▲ 9 stack frames were expanded.
async openRemoteImodel
C:/src/services/iModel/IModelService.ts:58
async getModelConnection
C:/src/components/iModel/IModelLoader.tsx:206
  203 |         allFrontstages.unshift({
  204 |             provider: defaultFrontstageProvider,
  205 |             default: true,
> 206 |         });
      | ^  207 |     }
  208 |     setFinalFrontstages(allFrontstages);
  209 | }, [frontstages, viewportOptions, viewState, defaultUiConfig]);
```

**Comment by aruniverse** [comment, marked as answer]
Take a look at this [announcement](https://github.com/imodeljs/imodeljs/discussions/2158) that should resolve your issue.

**Reply by imperidal** [reply]
Thanks that worked.  Do you know why I'm getting weird texts on my app?

![image](https://user-images.githubusercontent.com/38965496/133530319-534be9a5-d7eb-401c-af00-5c023c8ed281.png)

**Reply by aruniverse** [reply]
That's "normal", it's only shown in development (when you run `npm run start`). They should show up as normal text when you run a production build (`npm run build` and then serving the static files generated by the build) of your app.

**Reply by mgooding** [reply]
> That's "normal", it's only shown in development (when you run `npm run start`). They should show up as normal text when you run a production build (`npm run build` and then serving the static files generated by the build) of your app.

To elaborate on this, the development text intentionally includes non-ASCII characters to validate that it can be localized without issue.

**Reply by imperidal** [reply]
Perfect.  Thanks for the explanation.
